### ISSUE TITLE
Fire onshow event on all modals

### ISSUE THREAD

**Issue by batata004** [fomantic/Fomantic-UI#452]
Hello!

I have 30 modals on a page (each one is completelly different) however when I initiallize each modal using `$("#modal_id").modal(( "setting", { closable:false;...}))` on page load, the "mouse hangs" for as long as 500 to 1000ms. Also, using Google PageSpeed Insights, when I remove the initiallization of modals on pageload it gets saved 12 points on the final score, so clearly initiallizing that much modals impacts user experience.

So I ask you: is it possible to attach some kind of event to ONLY initiate the modals when the `.modal("show")` is called for the first time on each modal?

Cause I have hundreds of `.modal("show")` calls all over the place on my pages. Wheter a user clicks on a button, on a link, hover a image, scrolls past at certain point on a page.. Those `.modal("show")` calls are all over the place and it would too tedious/labour intensive to change every single `.modal("show")` to something like 

```
if (typeof initiated_modal_23 === "undefined") {
    
    initiated_modal_23  = true;
    $("#modal_id").modal(( "setting", { closable:false;...}));

}

$("#modal_id").modal("show");

```

For some reasom initializing modals with many content inside affects heavily page load. So it would be nice to know if there is some way to solve this problem.

OBS: I could defer the initialization of modals after 3 seconds (for example) from page load but if any `.modal("show")` gets executed before the initialization problems will happen, as you expect.

**Comment by lubber-de** [comment]
If all your modals share the same settings, you could set them globally

```javascript
$.fn.modal.settings.closable = false;
```

Other than that we would have to implement a new option like `instantiateOnShow`.

**Comment by batata004** [comment]
Thanks @lubber-de ! I tried this `$.fn.modal.settings.onShow = function() {alert('xx');}` but does not look to work, I mean, everytime I try to open a modal the alert is not being fired.

**Comment by ColinFrick** [comment]
@batata004 this works for me: https://jsfiddle.net/5ewq4tdy/
What's important is that you have to set the onShow before any modal is initialized.

**Comment by batata004** [comment]
@ColinFrick I was testing `$.fn.modal.settings.onShow ` after the modals have been initiated!  It indeed worked like charm as you said! Thanks!

**Comment by batata004** [comment]
@ColinFrick just one final question regarding this feature. After I do:

`$.fn.modal.settings.onShow = function () {...}`

Is it possible to remove this function after first run? I just need it to run once. I did this:

`$.fn.modal.settings.onShow = function () {

  if (typeof xxx === "undefined") {

      xxx = true;
      //MY INITIALIZATING CODE of modals come here

    console.log("aaa");
  }
console.log("bbb");
}`

When I open the first modal my console shows "aaa" and "bbb". After that, whenever I open a modal it always shows "bbb" on the console. Is it possible to "remove" this function that was added to the `onshow`? Something like `.one()` provided by jquery.
